lib/sorbet-coerce.rb

```diff
@@ -1,21 +1,21 @@
 # typed: strict
+require_relative 'configuration'
 require 'private/converter'
 require 'safe_type'
 
-module T
-  class CoercionError < SafeType::CoercionError; end
+module T::Coerce
+  class ValueError < SafeType::CoercionError; end
+  class ShapeError < SafeType::CoercionError; end
 
-  module Coerce
-    define_singleton_method(:[]) do |type|
-      Class.new(T::Private::Converter) do
-        define_method(:to_s) { "#{name}#[#{type.to_s}]" }
+  define_singleton_method(:[]) do |type|
+    Class.new(T::Private::Converter) do
+      define_method(:to_s) { "#{name}#[#{type.to_s}]" }
 
-        define_method(:from) do |args|
-          begin
-            T.send('let', send('_convert', args, type), type)
-          rescue TypeError
-            raise CoercionError.new(args, type)
-          end
+      define_method(:from) do |args|
+        begin
+          T.send('let', send('_convert', args, type), type)
```

Comment:
Hi! Could we convert `Configuration` into an extra `option`? Eg:

```
T::Coerce[Params].new.from({a: 'abc'}, raise_value_errors: false) # imo should default to true
```

This way it would be possible to decide when to raise the error.

Reply:
Great idea!